.github/workflows/deploy_staging.yml

```diff
@@ -0,0 +1,190 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/SchemaStore/schemastore/master/src/schemas/json/github-workflow.json
+---
+name: Deploy CloudFormation stacks for staging environment
+
+on:
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - unlabeled
+
+concurrency:
+    group: ${{ github.workflow }}-${{ github.ref }}
+    cancel-in-progress: true
+
+permissions:
+    contents: read
+    deployments: write
+
+env:
```

Comment:
A nice improvement would be to consolidate the env somewhere, but this isn't the worst thing just because it's pretty visible/clear what the env should look like for each workflow.